keps/sig-storage/177-volume-snapshot/README.md

```diff
@@ -231,24 +251,288 @@ Here are the changes since the original design proposal:
+      resources: ["volumesnapshotclasses"]
+      verbs: ["get", "list", "watch"]
+    - apiGroups: ["snapshot.storage.k8s.io"]
+      resources: ["volumesnapshotcontents"]
+      verbs: ["create", "get", "list", "watch", "update", "delete"]
+    - apiGroups: ["snapshot.storage.k8s.io"]
+      resources: ["volumesnapshots"]
+      verbs: ["get", "list", "watch", "update"]
+    - apiGroups: ["snapshot.storage.k8s.io"]
+      resources: ["volumesnapshots/status"]
+      verbs: ["update"]
+    - apiGroups: ["snapshot.storage.k8s.io"]
+      resources: ["volumesnapshotcontents/status"]
+      verbs: ["update"]
+```
+
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+  focusing mostly on:
+  - components listing and/or watching resources they didn't before
+    snapshot-controller and CSI external-snapshotter sidecar
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+    leader election
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Describe them, providing:
+  - API type: volumesnapshots, volumesnapshotcontents, and volumesnapshotclasses are CRDs
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+  If a cloud provider has implemented CSI snapshot functionality, it will be called when snapshot operations are triggered.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+  Describe them, providing:
+  - API type(s): New CRDs volumesnapshots, volumesnapshotcontents, and volumesnapshotclasses are added
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+  Think about adding additional work or introducing new steps in between
+  (e.g. need to do X to start a container), etc. Please describe the details.
+  This is a new feature so it will not increase time taken by existing operations.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  Things to keep in mind include: additional in-memory state, additional
+  non-trivial computations, excessive access to disks (including increased log
+  volume), significant amount of data sent and/or received over network, etc.
+  This through this both in small and large cases, again with respect to the
+  [supported limits].
+  This will add new snapshot API objects to the API server.
```

Comment:
what about the IO for the snapshot?  Is only supported for copy on write filesystems?

Reply:
IO depends on the underlying storage system.  We only provide a snapshot API at control layer.  Copy on write is typically used by snapshot technologies, but our snapshot API does not make any requirements on that because it depends on individual storage systems.

Reply:
> IO depends on the underlying storage system. We only provide a snapshot API at control layer. Copy on write is typically used by snapshot technologies, but our snapshot API does not make any requirements on that because it depends on individual storage systems.

Can you describe the risks in the doc here?  Perhaps something like: depending on the CSI driver snapshot implementation, taking a snapshot can drive significantly more IO on nodes.

Reply:
Some storage systems upload the snapshot to an object store after the snapshot is cut.  That could take a very long time.  I'll add a note for this in the KEP.